frontend/src/views/WheresMyGene/components/HeatMap/components/Chart/components/Chart/hooks/utils.ts

```diff
@@ -0,0 +1,196 @@
+import {
+  EChartsOption,
+  DatasetComponentOption,
+  ScatterSeriesOption,
+} from "echarts";
+import { ViewId } from "src/views/WheresMyGene/common/types";
+
+export interface CreateChartOptionsProps {
+  /**
+   * The data array to be visualized
+   * The data point object shape can be whatever you like, but it must be consistent with the `encode` option
+   * For example, if the data point shape is:
+   * {
+   *   geneIndex: 0,
+   *   cellTypeIndex: 0,
+   *   percentage: 0.5
+   * }
+   * and you want geneIndex to be encoded to x axis and cellTypeIndex to be encoded to y axis, then make sure your encode option is:
+   * encode: {
+   *   x: 'geneIndex',
+   *   y: 'cellTypeIndex'
+   * }
+   */
+  data: DatasetComponentOption["source"];
+  /**
+   * The data for the x axis
+   * For example:
+   * [{ value: "gene1", textStyle: { color: "red" } }, "gene2", "gene3"]
+   */
+  xAxisData: CategoryAxisData;
```

Comment:
Retyped this by manually recreating echarts' CategoryAxisOption type as `CategoryAxisData`, since it's not exported!